### PR TITLE
Blog Search List Clean Up

### LINKED ISSUE #42: Blog Search List Page
https://trello.com/c/o4qRgOF3/33-blog-search-list-page

RY: Flagging as Duplicate of "Blog API Call" card

 

- [ ] FE: Implement Search results

 

- [ ] FE: Implement routes for blog/ (all blogposts), blog/?topic=# (where # is topic ID number, or hyphenated-topic-name), and blog/?search-term

- [ ]  FE: Pass querystring to API endpoint

### PR DESCRIPTION
**Pull request by jessewoo**
WIP in progress. 
* Did more component-ization.
* Build a dummy json data object with HTML renderings of what blog post would be like. 
* Utilized components existing in the semantic library instead of rolling up our own styles. 
* Since results are not static but dynamically generated, the search bar is less of a filter but need to call api. With semantic search bar implemented, I was always getting a no results message found
* Used hooks instead of class component. 
 
Still need to hook up with api, work on pagination, work on sanitizing the data coming from response. 

@misunchoi - please take a look, ask me any questions you might have. 

Closes #42

### CODE REVIEW

File: frontend/src/components/FilterDropdown/FilterDropdown.jsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import React from 'react';
 import styled from 'styled-components';
 import { Dropdown } from 'semantic-ui-react';
 
+// QUESTION: Do we need this? We can use semantic select dropdown
```
Comment (misunchoi) [inline frontend/src/components/FilterDropdown/FilterDropdown.jsx:5]: I added a customized dropdown to match our design as much as I can, but we can use what semantic has already if they don't mind that

**Comment by misunchoi** [comment]
@jessewoo the overall structure looks good to me, thank you for working on it!
but are we not worrying about matching the figma design for now? I noticed small differences (like font styles) here and there but wasn't sure that was because the UI is still working on their design

**Comment by douglaschan32167** [comment]
The UX team has pretty much finalized their design (knock on wood) so we should try to match what's on  the figma as much as possible. We don't have that much time until code freeze though, so if it's a lot of work then we can skip it for now.

**Comment by jessewoo** [comment]
tbh, as a UI engineer, don't worry about fine tuning the styling because that can always change. And it's going to change once feedback hits. they don't like the color scheme, we gotta change that. they don't like the rounded input, gotta change. design is sort of icing, it's just changing css. Also, you can always push back and say this input styling doesn't match this input on another page, it's not uniform. And utilizing semantic UI components - that's already proven that the larger community buys into the styling. 

let's worry about getting the structure of the UI code first, how data is passed thru from the DB to what is displayed, what are the methods to get that done. We can fine tune styling during the period of code freeze and launch. Hope that make sense @misunchoi

**Comment by jessewoo** [comment]
@misunchoi - just to understand my thought process... in this process of refactoring the page, I've been pinging Doug back and forth about the data. What's coming back to the client, so it's going to be a big blob of data, now I got to transform it so that the component can use it. Is there data I'm missing that I need to let Doug know? What are the levers I can pull in terms of sorting and filtering like do we have capability to sort by latest? How am I going to do async request? What is isLoading state looking like, that's why I added in the spinner.

**Comment by misunchoi** [comment]
@jessewoo I see, I was playing around with the request and looks like the `query` param isn't working as expected, not sure if you ran into that issue too; but for the filtering and sorting the list I think we still need an endpoint for those. Same for the loading state.

**Comment by douglaschan32167** [comment]
@misunchoi for sorting, you can now add ?ordering=id or ?ordering=id or ?ordering=publish_at or ?ordering=-publish_at to your blogpostcontent api call to order based off of id or when they were published. See https://github.com/International-Mentorship-ISMP/ismp/blob/master/backend/docs/api-endpoints.md

That being said, I've been told that we will have ~4 blogposts at launch, so we could also just put off exposing the sort by dropdown until later.

File: frontend/src/pages/BlogSearchResults/BlogListPage.jsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  const fetchResultResponse = () => {
+    const queryParamSearch =
+      searchInputs.searchTerm.length > 0
+        ? `query='${searchInputs.searchTerm}'`
+        : '';
+
+    console.log(searchInputs);
+    const queryParamTopic =
+      searchInputs.topics !== 'all'
+        ? `topic_name='${searchInputs.topics}'`
```
Comment (douglaschan32167) [inline frontend/src/pages/BlogSearchResults/BlogListPage.jsx:72]: topic, not topic_name

**Comment by misunchoi** [comment]
@jessewoo I did the async call similar to how the CollegeCardList is doing; this should solve the loading state as well
```
const [blogList, setBlogList] = useState([]);
useEffect(() => {
    setIsLoading(true);
    const blogListResults = async () => {
      try {
        const blogListData = await requests
          .get('blogpostcontent')
          .then(response => response.results);
        setBlogList(blogListData)
        setIsLoading(false);
      } catch (error) {
        console.log(error);
      }
    };
    blogListResults();
  }, []);

  console.log(blogList)
```

File: frontend/src/components/FilterDropdown/FilterDropdown.jsx (hunk 1 above)
Comment (jessewoo) [reply frontend/src/components/FilterDropdown/FilterDropdown.jsx:5]: I'll just delete it for now. We can resurrect it later.

**Pull request merged**